DSCResources/MSFT_xIisFeatureDelegation/MSFT_xIisFeatureDelegation.psm1

```diff
@@ -0,0 +1,168 @@
+        [ValidateSet("Allow", "Deny")]
+        [String]$OverrideMode
+    )
+    
+    CheckIISPoshModule
+
+    [string]$oMode = GetOverrideMode -section $SectionName
+
+    if ($oMode -eq $null)
+    {
+        $ensureResult = "Absent";
+    }
+    else
+    {        
+        if ($oMode -eq $OverrideMode)
+        {
+            $ensureResult = "Present"
+        }
+        else
+        {
+            $ensureResult = "Absent";
+        }
+    }
+
+    # in case the section has not been found, $oMode will be $null
+    $getTargetResourceResult = @{SectionName = $SectionName
+                                OverrideMode = $oMode}
+
+    
+    return $getTargetResourceResult
+}
+
+######################################################################################
+# The Set-TargetResource cmdlet.
+# This function set the OverrideMode for a given section if not already correct
+######################################################################################
+function Set-TargetResource
+{
+    param
+    (    
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [String]$SectionName,
+
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateSet("Allow", "Deny")]
+        [String]$OverrideMode
+    )
+
+    CheckIISPoshModule
+    [string]$oMode = GetOverrideMode -section $SectionName
+
+
+    if ($oMode -eq "Allow" -and $OverrideMode -eq "Deny")
+    {
+         Set-webconfiguration -Location "" -Filter "/system.webServer/$SectionName" -PSPath "machine/webroot/apphost" -metadata overrideMode -value Deny
+         Write-Verbose($($LocalizedData.ChangedMessage) -f $SectionName,"Deny")
+    }
+    elseif ($oMode -eq "Deny" -and $OverrideMode -eq "Allow")
+    {
+         Set-webconfiguration -Location "" -Filter "/system.webServer/$SectionName" -PSPath "machine/webroot/apphost" -metadata overrideMode -value Allow
+         Write-Verbose($($LocalizedData.ChangedMessage) -f $SectionName,"Allow")
+    }
+}
+
+######################################################################################
+# The Test-TargetResource cmdlet.
+# This will test if the given section has the required OverrideMode
+######################################################################################
+function Test-TargetResource
+{
+    [OutputType([System.Boolean])]
+    param
+    (    
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [String]$SectionName,
+
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateSet("Allow", "Deny")]
+        [String]$OverrideMode
+    )
+
+    CheckIISPoshModule
+
+    $DesiredConfigurationMatch = $false;
+    [string]$oMode = GetOverrideMode -section $SectionName
+
+    if ($oMode -eq $OverrideMode)
+    {
+        # this is the only case where we have our desired state
+        $DesiredConfigurationMatch = $true;
+    }
+    elseif ($oMode -ne $null -and $OverrideMode -ne $oMode)
```

Comment:
I'd probably just change the language on the else statement to indicate it doesn't match. Honestly given that it doesn't match does it really matter that much if it's because it doesn't exist?

If you want it to be specific in why its not matching then I would have the verbose statement in the elseif stating why. Otherwise I would chuck the elseif/else as its IMO unnecessary.